dbms/src/Functions/FunctionsStringSearch.h

```diff
@@ -20,9 +20,89 @@
 #include <DataTypes/DataTypesNumber.h>
 #include <Functions/FunctionHelpers.h>
 #include <Functions/IFunction.h>
+#include <Common/Exception.h>
+#include <Common/StringUtils/StringUtils.h>
+
+#include <string_view>
 
 namespace DB
 {
+using Chars_t = ColumnString::Chars_t;
+using Offsets = ColumnString::Offsets;
+
+inline void lowerOneString(Chars_t & str_container, size_t pos, size_t size)
```

Comment:
Can it work for different tidb collations such as unicode ci？

Reply:
> Can it work for different tidb collations such as unicode ci？

According to pm, collation is ignored when using `ilike` and we only need to lower the ascii.